crates/rslint_parser/src/syntax/function.rs

```diff
@@ -395,13 +395,37 @@ pub(super) fn is_at_async_function(p: &Parser, should_check_line_break: LineBrea
     }
 }
 
+pub(super) enum Ambiguity {
+    Allow,
+    Disallow,
+}
```

Comment:
Nit: 

```suggestion
pub(super) enum Ambiguity {
    Allowed,
    Disallowed,
}
```

It sounds better in my head

Reply:
Would you mind add a comment explaining what kind of ambiguity we have to deal with, please? Without the context of the PR, it might be difficult to get it right away

Reply:
I documented the `ambiguity` in the method parsing the arrow function. The enum itself could be reused in other places as well. But I can add a little more documentation.

Reply:
That's the thing, too. I read the comment but I didn't understand what's the actual ambiguity. I needed to check the PR again to remember it, because the description of the PR gives more context. 

I am sure if I will stumble across this code again, I won't remember what's the ambiguity about.

Reply:
I added some more documentaiton. I hope that helps.